kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,22 +105,156 @@ the stored credentials.
+			if cmd.Flag("image").Changed == false && cmd.Flag("blueprint").Changed == false {
+				util.LogAndExit("Please specify either the blueprint file or the image path")
+			}
+			if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+				util.LogAndExit("%s", "Cannot specify both blueprint and image in command")
+			}
+			if cmd.Flag("image").Changed == true {
+				fmt.Printf("Deploying image %s\n", cmd.Flag("image").Value.String())
+			} else if cmd.Flag("blueprint").Changed == true {
+				fmt.Printf("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
+			}
+		},
+		Run: deployOpts.deployImpl,
 	}
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.blueprintFilePath, "blueprint", "b", "", "blueprint file path.")
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.imagePath, "image", "i", "", "docker image path.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.username, "username", "u", "", "docker registry username.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicasByStreamletName, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	contents, err := fileutils.GetFileContents(blueprintFile)
+	contents = strings.TrimSpace(contents)
+	if err != nil {
+		util.LogAndExit("Failed to fetch blueprint contents from %s", blueprintFile)
+	}
+
+	blueprint, err := verify.VerifyBlueprint(contents)
+	if err != nil {
+		util.LogAndExit("Blueprint verification failed. Error: %s", err.Error())
+	}
+
+	util.PrintSuccess("Blueprint verified for application %s .. Proceeding with deployment\n", blueprint.GetName())
+
+	applicationSpec, pulledImages, err := deploy.CreateApplicationSpecFromBlueprintAndImages(blueprint, opts.replicasByStreamletName)
+	if err != nil {
+		util.LogAndExit("Failed to create cloudflow application spec from blueprint and images: %s", err.Error())
+	}
+
+	namespace := applicationSpec.AppID
+	k8sClient, k8sErr := k8s.GetClient()
```

Comment:
Might want to move the connect to k8s parts all the way up (getting the clients), especially if the command currently  takes very long (better to fail fast on not getting to the cluster first, then do long process)